File: app/src/main/java/com/arturogutierrez/openticator/di/module/ActivityModule.java
package com.arturogutierrez.openticator.di.module;

import android.app.Activity;
import com.arturogutierrez.openticator.di.PerActivity;
import dagger.Module;
import dagger.Provides;

@Module
public class ActivityModule {
  private final Activity activity;

  public ActivityModule(Activity activity) {
    this.activity = activity;
  }

  @Provides
  @PerActivity
  Activity activity() {
    return activity;
  }
}

File: app/src/main/kotlin/com/arturogutierrez/openticator/di/module/ActivityModule.kt
package com.arturogutierrez.openticator.di.module

import android.app.Activity
import com.arturogutierrez.openticator.di.PerActivity
import dagger.Module
import dagger.Provides

@Module
class ActivityModule(private val activity: Activity) {

  @Provides
  @PerActivity
  fun activity(): Activity = activity
}
